src/primitives/transaction.h

```diff
@@ -250,6 +250,7 @@ class CTransaction
     unsigned int CalculateModifiedSize(unsigned int nSize = 0) const;
 
     bool IsCoinBase() const { return (vin.size() == 1 && vin[0].prevout.IsNull()); }
+    bool IsProofBase() const { return (vin.size() > 1 && vin[0].prevout.IsNull() && vout.empty()); }
```

Comment:
Do we want to require that `vin.size() > 1`? By doing that, we are requiring proofs to have at least one predecessor. But at the beginning of the mining interval, this will not be the case.

Reply:
it is a requirement that vout be empty for a proofbase. a coinbase can be valid with an empty vout. There needs to be some strong distinction between a coinbase and a proofbase. if you have no predecessors then a proofbase would have to push 2 prevouts that are null instead of 1 to be distinguishable from a coinbase.

Reply:
> a proofbase would have to push 2 prevouts that are null instead of 1 to be distinguishable from a coinbase.
I see, that works for me. But why is it so important that a proofbase be invalid as a coinbase? The protocol will not treat transactions that appear in a subblock as a coinbase in any case, right?

Reply:
It wouldnt but a clear distinction is imo cleaner and makes it harder to make a mistake